### DISCUSSION TITLE
Thinking of using wiktionary.org as source for Free Dictionary API. What are your thoughts?

### DISCUSSION THREAD

**Discussion by meetDeveloper** [meetDeveloper/freeDictionaryAPI#101] [Q&A]
**tl;dr.**
- Will start using Wiktionary as source for Dictionary API
- API will only support English for a while and will later bring support for Spanish and French languages.
- API structure will be maintained and changes if any will only be additive in nature.
- API can be then used commercially as well as Wiktionary entries are dual-licensed to the public under both the [Creative Commons Attribution-ShareAlike 3.0 Unported License (CC-BY-SA)](https://creativecommons.org/licenses/by-sa/3.0/) and the [GNU Free Documentation License (GFDL)](https://www.gnu.org/licenses/fdl-1.3.en.html), more information about this [here](https://en.wiktionary.org/wiki/Wiktionary:Copyrights).

Hello,

I am planning to start using Wiktionary as source for Dictionary API. This will be an replacement to the current source, and due to this languages other than English will have to be dropped. Though for French and Spanish languages I will try to bring support soon.

The original texts of Wiktionary entries are dual-licensed to the public under both the [Creative Commons Attribution-ShareAlike 3.0 Unported License (CC-BY-SA)](https://creativecommons.org/licenses/by-sa/3.0/) and the [GNU Free Documentation License (GFDL)](https://www.gnu.org/licenses/fdl-1.3.en.html), more information about this [here](https://en.wiktionary.org/wiki/Wiktionary:Copyrights). Wiktionary sources information from [these ](https://en.wiktionary.org/wiki/Wiktionary:Public_domain_sources) public domain sources.

Using Wiktionary as a source will help developers using my API a lot as they won't have to worry about using my API for commercial usage etc.

Currently Wiktionary for English language is pretty solid and we can rely on that, I have compared many entries and see no problem in using that as our source for English language.

The only downside I see is that the API will be exclusively for English language as for other languages as Wiktionary lags behind,      but you can expect other languages to be added as soon as possible. Given that majority of users of this API are English speaking users, this change will affect relatively fewer users. 

Users that need to have dictionary for other languages for their own personal use can use google search, google provides dictionary for many languages, I will add a detailed write up regarding how to search meaning for different languages on google later. In case you need an API to power your app, I suggest you to look at the Oxford Dictionary API, AFAIK they have Dictionary API that supports many languages.

I will maintain the structure of the API same and changes if any will only be additive in nature.

Thanking You
meetDeveloper
